lib/ClangImporter/ImportDecl.cpp

```diff
@@ -437,16 +437,20 @@ getSwiftStdlibType(const clang::TypedefNameDecl *D,
     M = Impl.getStdlibModule();
   else
     M = Impl.getNamedModule(SwiftModuleName);
-  if (!M) {
-    // User did not import the library module that contains the type we want to
-    // substitute.
-    *IsError = true;
-    return std::make_pair(Type(), "");
-  }
 
   Type SwiftType = Impl.getNamedSwiftType(M, SwiftTypeName);
+
+  if (!SwiftType && CTypeKind == MappedCTypeKind::CGFloat) {
+    // Fall back to looking for CGFloat in Core Graphics.
+    M = Impl.getNamedModule("CoreGraphics");
+    SwiftType = Impl.getNamedSwiftType(M, SwiftTypeName);
+  }
+
   if (!SwiftType && !CanBeMissing) {
```

Comment:
There's a slight behavior change here that might explain the failure in `module-cache-diagnostics.swift`: when there is no module and `CanBeMissing`, the old code would unconditionally return `std::make_pair(Type(), "")`. The new code will return `std::make_pair(Type(), SwiftTypeName)`.

Reply:
Thanks for catching that. I'll add the module check back